cv19ResSupportV3.Tests/V3/UseCase/CreateHelpRequestCallUseCaseTest.cs

```diff
@@ -27,23 +28,19 @@ public void SetUp()
         public void ExecuteWithValidIdSavesRequestToDatabase()
         {
             int id = 1;
-            var expectedResponse = new HelpRequestCallCreateResponse
-            {
-                Id = id
-            };
-            _mockHelpRequestCallGateway.Setup(s => s.CreateHelpRequestCall(id, It.IsAny<HelpRequestCallEntity>())).Returns(id);
-            var dataToSave = new Fixture().Build<HelpRequestCall>().Create();
+            _mockHelpRequestCallGateway.Setup(s => s.CreateHelpRequestCall(id, It.IsAny<CreateHelpRequestCall>())).Returns(id);
+            var dataToSave = new Fixture().Build<CreateHelpRequestCall>().Create();
             var response = _classUnderTest.Execute(id, dataToSave);
-            response.Should().BeEquivalentTo(expectedResponse);
+            response.Should().Be(id);
         }
 
         [Test]
         public void ExecuteWithInvalidIdReturnsNull()
```

Comment:
I think the test name doesn't quite match the assertion. I also have question - does a 0th Id exist, or do they start being enumerated from 1? (If so, maybe it's better to return null?)

Reply:
So I decided to look at this in more detail. There's a bigger issue here than it first appears.

The `HelpRequestCallGateway.CreateHelpRequestCall` function does several strange things:
* It returns `0` when the comand is null.
  * It should be throwing an `IllegalArgumentException`
  * It should never be null, and we should use the nullity compiler option to prevent it being allowed to be null.
* It throws an `InvalidOperationException` if the HelpRequest cannot be found - this is actually OK however:
  * Consumers of this gateway function will not know that an IOE is to be expected as it's not documented
  * C# doesn't provide a way of forcing consumers to deal with exceptions (unlike Java with its checked exceptions.). This leads us to a scenario where we need intimate knowledge of the logic of the gateway, we need to look at the code (or doc..). If the gateway behaviour changes later, the mock in this test will not change behaviour. There's simply no contract.

An alternative would be for the Gateway to return a response object, instead of just an ID. This way the gateway can be explicit about whether it was successful, or not, and if not, what was the issue. See below for an example where a gateway can provide that extra data to the usecase, to enable the usecase to do something off the back of a failure.

https://github.com/madetech/nhs-virtual-visit/blob/8e0c642e5779316f88d33f1f6993b5115df6cbcf/src/gateways/retrieveTrustById.js#L4-L32

Reply:
This is out of scope for this PR as it's already quite large - we can pick this up separately.